modules/internal/ChapelArray.chpl

```diff
@@ -5177,12 +5148,12 @@ module ChapelArray {
     return result;
   }
 
-  proc chpl__fixupConstDomain(dom: domain, definedConst: bool) 
+  inline proc chpl__fixupConstDomain(dom: domain, definedConst = true)
```

Comment:
I wouldn't make either of the changes here.

- `inline` is a directive in Chapel and not a hint like it is in C. So, when you do this, this function will be inlined no matter what. It may look OK here, but `_value` is another inlined `proc`, so it'll be inlined, too. For this specific case, all this inlining will not generate a huge code, but it is a maintenance burden. If any of the downstream functions in this inline chain changes, it can create unexpected code bloat. Given that we already have a problem with compilation speed, and we don't yet know domain creation to be a particular bottleneck, I'd remove this `inline` until we have a concrete reason to add it back. Our module code has not been particularly attentive of this, but we are hoping to change it a bit and started doing that (reducing the number of `inline`s) in the last release cycle.
- Default arg for `definedConst`: I can see that there are bunch of new use cases for this proc with this PR, and having this default value can help. But in our internal module code, I am in favor of verbosity instead of ease of typing. Having default arguments can cause bugs to sneak in silently. For example, if this formal didn't have default value, compiler would have complained about the case that I commented on above, and that would have forced you to add the `definedConst` argument.

Reply:
While the reason for avoiding the default value for `definedConst` applies to both overloads, I suggest keeping the `inline` annotation on the other overload whose body is empty.

Reply:
The `inline` was intentional as a "forced inline". I don't like inline "hints"; I consider those implicit. (In languages like C++, inline also has linkage consequences unrelated to whether inlining actually occurs.) Since the function performs either a no-op or sets a single field in a record, it should probably be inlined, unless it interferes with the debugging of modules and/internal functions.

The default argument was not just to save typing; the function name seems to imply setting const by default. I would rather remove the default argument and make the call to the function conditional based on an if-statement, instead of passing in a Boolean.

In my experience, passing Booleans to functions costs more than it's worth, since you're passing a single bit through a heavy ABI, and then the bit may just control a branch which is less predictable in a general procedure than if the branch is predicted at each caller site.

So I'd remove the second argument and only call the internal function if the condition is true. Since the routine would be a no-op or a single statement, it should probably be inlined.